src/pybind/ceph_volume_client.py

```diff
@@ -710,6 +710,23 @@ def _metadata_set(self, path, data):
     def _lock(self, path):
         @contextmanager
         def fn():
+            while(1):
+                fd = self.fs.open(path, os.o_creat, 0755)
+                self.fs.flock(fd, fcntl.lock_ex, self._id)
+
+                # The locked file could have been been deleted e.g, by an
+                # another manila share instance. Perform checks to ensure that
+                # the lock is not applied on an unlinked file.
+                try:
+                    statbuf = self.fs.stat(path)
+                except cephfs.objectnotfound:
+                    self.fs.close(fd)
+                    continue
+
+                fstatbuf = self.fs.fstat(fd)
+                if statbuf.st_ino == fstatbuf.st_ino:
+                    break
+
             fd = self.fs.open(path, os.O_CREAT, 0755)
```

Comment:
You'd want to remove this open/flock call, as it's already been done in the loop above, right?